sips/sip-015/sip-015-network-upgrade.md

```diff
@@ -0,0 +1,1582 @@
+contract, which would then prohibit them from being published.  However, there
+are some classes of check errors (such as the one described above) where this is
+infeasible.
+
+This SIP proposes treating all check errors that arise during
+transaction-processing as runtime errors.  Transactions that encounter check
+errors would be mined, and their fees would be collected, but the transaction's
+payload would never be applied to the chainstate.
+
+This change benefits both miners and users.  With this change, miners will get paid for
+processing a transaction regardless of how it would fail when executed.  Notably, this change
+does _not_ prevent the user's wallet from executing a more-rigorous analysis
+pass on any transactions the user sends.
+
+This change also helps users because it means that transactions that encounter
+check errors will get cleared from the mempool without any further action on
+their parts.  Today, users must either replace-by-fee (RBF) the problematic
+transaction.
+
+# Related Work
+
+Most blockchains regularly execute coordinated breaking changes
+to add new features or behaviors.  Prominent examples include Ethereum [2] and
+Tezos [3].  Indeed, Stacks has already gone through one breaking change [4] via
+its SIP process.
+
+Stacks strikes a balance between developer-coordinated breaking changes (like
+Ethereum) and self-amending ledgers (like Tezos).  Like Ethereum, changes in
+Stacks are sufficiently disruptive that for now, upgrades must be coordinated
+out-of-band and are led by developers.  But unlike Ethereum, the Stacks SIP
+process [5] has well-defined constraints on developers' ability to effect
+changes unilaterally -- in particular, Stacks' governance system requires
+independent reviewers to authorize breaking changes, and in the case of breaking
+changes, requires (through precedent) users to explicitly consent to them
+in-band through coin votes.
+
+# Backwards Compatibility
+
+This SIP retains full backwards-compatibility with the Stacks 2.05 chainstate.
+All Clarity features that are supported today will continue to be supported.
+However, developers publishing new Clarity 1 contracts will need to use the new
+contract-publish transaction and explicitly indicate that they want to use
+Clarity 1.  All smart contracts published with the standard contract-publish
+transaction will default to using Clarity 2 rules.
+
+Due to the introduction of new reserved keywords in Clarity 2, it will not be
+possible to publish an already-written contract as a Clarity 2 contract if it
+uses these reserved words for any other purpose.  However, already-deployed
+contracts are not affected -- they were published under Clarity 1 rules, and
+will continue to be usable as-is.
+
+The read-only contract calls in `pox` will continue to be available to
+contracts.  However, it will no longer be possible to call public functions in
+`pox`.  All future Stacking operations must be done through `pox-2`.
+
+# Activation 
```

Comment:
Tagging @Hero-Gamer, @jennymith, and @radicleart to make sure this section accurately describes what we've discussed in the past.

Reply:
Pretty important dates - might be worth having them at the very top as 'proposed activation window' 

Would also call for a table here - with a column for each actor in the process (stacker, non-stacker, miner) and rows for actions they need to take and dates when they need to be completed by. 

This may be best placed in the 'Activation Status' sections that follows this - I leave it to your best judgement

Reply:
Good idea!  Will copy it to the Abstract.

Reply:
I was going to create a flowchart instead of a table.  Just haven't gotten around to it yet.